src/renderer/PresetsBar.tsx

```diff
@@ -113,29 +114,37 @@ const PresetsBar = () => {
     'Jazz',
   ]);
 
-  const handleCreatePreset = (prev: string[]) => {
-    if (prev.indexOf(presetName) !== -1) {
-      // TODO: overwrite preset values for the existing preset
-    } else {
-      // TODO: add code for creating a preset
-      setSelectedPresetName(presetName);
+  const handleCreatePreset = async (prev: string[]) => {
+    await savePreset(presetName);
+
+    // If we are creating a new preset and not just updating an existing one
+    if (prev.indexOf(presetName) === -1) {
+      await setSelectedPresetName(presetName);
```

Comment:
Since `setSelectedPresetName` doesn't return a `Promise`, the `await` keyword isn't necessary here. The existence of the `await` in line 188 means that the code below it will be treated as if it were wrapped in a `.then(() => { ... })`.